bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -225,4 +249,28 @@ private Sitemap getSitemap(String sitemapName) {
         return null;
     }
 
+    @Override
+    public void modelChanged(String modelName, EventType type) {
+
+        if (type != EventType.MODIFIED || !modelName.endsWith("." + SITEMAP_SUFFIX)) {
+            return; // we process only sitemap modifications here
+        }
+
+        String changedSitemapName = modelName.substring(0, modelName.indexOf("."));
```

Comment:
ouch, no!

That's gonna be better:

```java
String changedSitemapName = StringUtils.removeEnd(name, SITEMAP_SUFFIX)
```

(which is what [SitemapProviderImpl](https://github.com/eclipse/smarthome/blob/3ae5d7b1b284f2bde6199bfe8dad950fd38cae39/bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java#L74) is doing too)